### ISSUE TITLE
The “Label volume editor” process fails to save the file header

### ISSUE THREAD

**Issue by ylep** [brainvisa/axon#39]
The `referentials` and `transformations` header fields should be preserved when the edited ROI is saved by the “Label volume editor” (`AnaLabelVolumeEditor`) process.

(Bug found in BrainVISA 4.6.1)

**Comment by ylep** [comment]
This bug is not fixed in the current `master` branch: the headers fail to be preserved correctly during the volume -> graph -> volume round-trip conversion.

**Comment by denisri** [comment]
Anatomist does write transformation information correctly. Maybe the problem is in the brainvisa process then. I'll check that.

**Comment by ylep** [comment]
For what it's worth I have implemented a quick-and-dirty patch:

```patch
diff --git a/brainvisa/toolboxes/tools/processes/roi/AnaLabelVolumeEditor.py b/brainvisa/toolboxes/tools/processes/roi/AnaLabelVolumeEditor.py
index 00430b8..828d2a0 100644
--- a/brainvisa/toolboxes/tools/processes/roi/AnaLabelVolumeEditor.py
+++ b/brainvisa/toolboxes/tools/processes/roi/AnaLabelVolumeEditor.py
@@ -35,7 +35,7 @@ from brainvisa.processes import *
 from brainvisa import anatomist
 from brainvisa.processing.qtgui.neuroProcessesGUI import mainThreadActions
 
-name = 'Label volume editor'
+name = 'Label volume editor (2020-05-20 hotfix by yleprince)'
 roles = ('editor',)
 userLevel = 0
 
@@ -138,6 +138,20 @@ def save_roi(self, message=None):
                            self.fgraphbase + '.data', 'roi_Volume'), '-o',
                        self.label_volume, '-g', -1, '-n', val)
 
+        # Restore the header of the label volume
+        from soma import aims
+        label_vol = aims.read(self.label_volume.fullPath())
+        label_vol_header = label_vol.header()
+        for name, value in six.iteritems(self.label_volume_header):
+            if name in ['volume_dimension']:
+                # These attributes cannot be set
+                continue
+            try:
+                label_vol_header[name] = value
+            except AttributeError as e:
+                context.warning('Cannot restore attribute {0}: {1}'.format(attr_name, e))
+            aims.write(label_vol, self.label_volume.fullPath())
+
 
 def execution(self, context):
     a = anatomist.Anatomist()
@@ -164,6 +178,8 @@ def execution(self, context):
     fgraphbase = os.path.join(tmpdir.fullPath(), 'finalgraph')
     finalgraph = fgraphbase + '.arg'
 
+    label_volume_header = aimsFileInfo(mask.fullPath())
+
     context.system('AimsGraphConvert',
                    '-i', mask.fullPath(),
                    '-o', voigraph,
@@ -193,6 +209,7 @@ def execution(self, context):
 
     self.context = context
     self.voigraph = voigraph
+    self.label_volume_header = label_volume_header
     self.window = windownum
     self.fgraphbase = fgraphbase
     self.finalgraph = finalgraph
```

I don't like the fact that it uses `aims` directly in the BrainVISA process though. The proper fix would be to preserve the header correctly during every conversion. @denisri I thought that we had fixed this a few months ago by loading the `.minf` associated with `.arg`, but apparently that was not enough.

**Comment by denisri** [comment]
It's within AimsGraphConvert, used to convert the ROI image into a ROI graph, which loses this information. More precisely the aims Reader used to read a volume (.nii) as a graph did not preserve the volume header. I'm fixing that. And it also needs passing a bit more information to the writer.
Your fix is an interesting temporary fix for the users who don't compile the C++ code until we release a new version.

**Comment by denisri** [comment]
Should be fixed by brainvisa/aims-free@67f986f0c9e6a09b59e62fb8603e66f9490d0aa2.

**Comment by ylep** [comment]
I am trying to test if the header issue is fixed, but now I cannot open the ROI graphs anymore:

```
/tmp/bv_vNd01H/voigraph.arg : unsupported data type Graph / S16
Could not read object in file /tmp/bv_vNd01H/voigraph.arg !
```

Could it be that https://github.com/brainvisa/aims-free/commit/67f986f0c9e6a09b59e62fb8603e66f9490d0aa2 introduced an incompatibility  that prevents Anatomist from reading the newly written graphs?

**Comment by ylep** [comment]
@denisri I can confirm that graphs written by the new writer (after https://github.com/brainvisa/aims-free/commit/67f986f0c9e6a09b59e62fb8603e66f9490d0aa2) cannot be loaded in Anatomist, while graphs written by the old one work just fine.

**Comment by ylep** [comment]
The issue is due to the `data_type` field of the Minf header: when I remove it, it can be loaded just fine.

**Comment by ylep** [comment]
@denisri I am not sure what is the better approach here:

1. specifically exclude the `data_type` field when writing a Graph (maybe also `bits_allocated`, `object_type`, `preferred_data_type`, maybe others)
2. Ignore these attributes when reading a Graph

Maybe combine both?

**Comment by denisri** [comment]
OK, I'm fixing it (I had added all volume header attributes directly in the graph at first, which worked but was not clean, then switched to the "header" property which holds the minf attributes, which is cleaner, but... didn't test the reload situation then... Sorry.

**Comment by denisri** [comment]
This should be fixed by brainvisa/aims-free@ecb1b437664a61d81ab7597acc680bdc368c2539.

**Comment by ylep** [comment]
Presumably fixed in the next release, closing the issue.
